fulltext/util.py

```diff
@@ -108,3 +113,39 @@ def run(*cmd, **kwargs):
 def warn(msg):
     warnings.warn(msg, UserWarning)
     LOGGER.warning(msg)
+
+
+def is_windows():
+    """True if the platform is Windows."""
+    return 'win' in sys.platform
+
+
+def is_windows64():
+    """
+    Determine if platform is 64 bit Windows.
+    """
+    return is_windows() and 'PROGRAMFILES(X86)' in os.environ
+
+
+if is_windows():
+    # Help the magic wrapper locate magic1.dll, we include it in bin/bin{32,64}
+    bindir = 'bin64' if is_windows64() else 'bin32'
+
+    os.environ['PATH'] += str(os.pathsep + pathjoin(BASE_PATH, 'bin', bindir))
+
+    # Then instantiate our own Magic instance so we can tell it where the
+    # magic file lives.
+    try:
+        from magic import Magic
+
+        magic = Magic(mime=True,
+                      magic_file=pathjoin(BASE_PATH, 'bin', 'magic'))
+    except Exception:
+        warn('Magic is unavailable, type detection degraded')
+        # If all else fails...
+        magic = None
+
+else:
+    # On linux things are simpler. Linter disabled for next line since we
+    # import here for export.
+    import magic  # NOQA
```

Comment:
For all of this to work, there are some binary files that have to be packaged in `setup.py` and `MANIFEST.in` I am OK with you omitting this for now. It took me a while to get all of this working in duster.